File: R/inst/examples/prostate.R
library(easyml) # https://github.com/CCS-Lab/easyml

# Load data
data("prostate", package = "easyml")

# Settings
.n_samples <- 50
.n_divisions <- 50
.n_iterations <- 2
.n_core <- 1

results <- easy_random_forest(prostate, "lpsa", 
                              n_samples = .n_samples, n_divisions = .n_divisions, 
                              n_iterations = .n_iterations, random_state = 1, n_core = .n_core)

results$plot_variable_importances
results$plot_predictions_single_train_test_split_train
results$plot_predictions_single_train_test_split_test
results$plot_model_performance_train
results$plot_model_performance_test

# Analyze data
results <- easy_glmnet(prostate, "lpsa", 
                       n_samples = .n_samples, n_divisions = .n_divisions, 
                       n_iterations = .n_iterations, random_state = 1, n_core = .n_core, 
                       model_args = list(alpha = 1, nlambda = 200))
results$plot_coefficients
results$plot_predictions_single_train_test_split_train
results$plot_predictions_single_train_test_split_test
results$plot_model_performance_train
results$plot_model_performance_test

# glinternet_results <- easy_glinternet(prostate, "lpsa", 
#                                       n_samples = .n_samples, n_divisions = .n_divisions, 
#                                       n_iterations = .n_iterations, random_state = 1, n_core = .n_core)

neural_network_results <- easy_neural_network(prostate, "lpsa", 
                                              preprocess = preprocess_scale, 
                                              measure = measure_r2_score, 
                                              n_samples = .n_samples, n_divisions = .n_divisions, 
                                              n_iterations = .n_iterations, random_state = 1, n_core = .n_core, 
                                              model_args = list(size = 50, decay = 1))

library(darch)
deep_neural_network_results <- easy_deep_neural_network(prostate, "lpsa", 
                                                        preprocess = preprocess_scale, 
                                                        measure = measure_r2_score, 
                                                        n_samples = .n_samples, n_divisions = .n_divisions, 
                                                        n_iterations = .n_iterations, random_state = 1, n_core = .n_core)

model_args <- list(size = 5, linout = TRUE, trace = FALSE)
g <- easy_avNNet(prostate, "lpsa", 
                 preprocess = preprocess_scale,
                 n_samples = 10,  n_divisions = 10, 
                 n_iterations = 10, 
                 random_state = 12345, n_core = 1, 
                 model_args = model_args)

File: Python/examples/prostate.py
from easymlpy import glmnet, random_forest, support_vector_machine
from easymlpy.datasets import load_prostate


# Settings
n_samples = 50
n_divisions = 50
n_iterations = 2

# Load data
prostate = load_prostate()

# Analyze data
output = support_vector_machine.easy_support_vector_machine(prostate, 'lpsa',
                                                            random_state=1, progress_bar=True, n_core=1,
                                                            n_samples=50, n_divisions=50, n_iterations=2)

output.plot_predictions_single_train_test_split_train()
output.plot_predictions_single_train_test_split_test()
output.plot_model_performance_train()
output.plot_model_performance_test()

# Analyze data
output = random_forest.easy_random_forest(prostate, 'lpsa',
                                          random_state=1, progress_bar=True, n_core=1,
                                          n_samples=50, n_divisions=50, n_iterations=2,
                                          model_args={'n_estimators': 10})
output.plot_variable_importances()
output.plot_predictions_single_train_test_split_train()
output.plot_predictions_single_train_test_split_test()
output.plot_model_performance_train()
output.plot_model_performance_test()

# Analyze data
output = glmnet.easy_glmnet(prostate, 'lpsa',
                            random_state=1, progress_bar=True, n_core=1,
                            n_samples=50, n_divisions=50, n_iterations=2,
                            model_args={'alpha': 1, 'n_lambda': 200})
output.plot_coefficients()
output.plot_predictions_single_train_test_split_train()
output.plot_predictions_single_train_test_split_test()
output.plot_model_performance_train()
output.plot_model_performance_test()
